lib/api/core/plugins/pluginsManager.js

```diff
@@ -634,24 +677,31 @@ class PluginsManager {
    * @throws {PluginImplementationError} If strategies registration fails
    */
   _initAuthentication (plugin) {
-    const errorPrefix = `Unable to inject authentication strategies from plugin "${plugin.name}": `;
+    const errorPrefix = `[${plugin.name}]:`;
 
-    if (!isObject(plugin.object.strategies) || Object.keys(plugin.object.strategies).length === 0) {
-      throw new PluginImplementationError(`${errorPrefix}The exposed "strategies" plugin property must be a non-empty object`);
+    if (!_.isPlainObject(plugin.object.strategies) || _.isEmpty(plugin.object.strategies)) {
+      throw new PluginImplementationError(`${errorPrefix} the exposed "strategies" plugin property must be a non-empty object`);
     }
 
-    Object.keys(plugin.object.strategies).forEach(strategyName => {
-      const strategy = plugin.object.strategies[strategyName];
-
-      try {
-        this.registerStrategy(plugin.name, strategyName, strategy);
+    // @todo the _.isNil test need to be removed as soon as the
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)